base/compiler/ssair/passes.jl

```diff
@@ -1096,41 +1125,32 @@ function cfg_simplify!(ir::IRCode)
             ms = merged_succ[ms]
         end
     end
+
+    # Compute statement indices the new blocks start at
     bb_starts = Vector{Int}(undef, 1+length(result_bbs_lengths))
     bb_starts[1] = 1
     for i = 1:length(result_bbs_lengths)
         bb_starts[i+1] = bb_starts[i] + result_bbs_lengths[i]
     end
-    # Figure out the pred and succ lists for each basic block in our merged result
-    cresult_bbs = let result_bbs = result_bbs,
-                      merged_succ = merged_succ,
-                      merge_into = merge_into,
-                      bbs = bbs,
-                      bb_rename_succ = bb_rename_succ
```

Comment:
Removing the `let` block prohibits inference of the types of the captured references, which could be a large performance penalty on the inner loops here.

Reply:
Good catch, I remember adding this let block for that very reason